sdk/cognitivelanguage/Azure.AI.Language.QuestionAnswering/tests/Samples/Sample5_UpdateKnowledgeSources.cs

```diff
@@ -46,8 +46,12 @@ public void KnowledgeSources()
                         }
                 });
 
-            Operation<BinaryData> updateSourcesOperation = client.UpdateSources(waitForCompletion: false, testProjectName, updateSourcesRequestContent);
+            Operation<BinaryData> updateSourcesOperation = InstrumentOperation(client.UpdateSources(waitForCompletion: false, testProjectName, updateSourcesRequestContent));
+#if SNIPPET
             updateSourcesOperation.WaitForCompletion();
+#else
+            await updateSourcesOperation.WaitForCompletionAsync();
+#endif
```

Comment:
You can't just keep this method sync and call the sync version of `WaitForCompletion()`? If not necessary, I'd just keep it simple / easy to maintain and easier to understand if a user views the original sample source (this).

Reply:
Unfortunately this runs into the timeout issue when running the test in playback. Using the async version of `WaitForCompletionAsync()` was part of the fix Christopher suggested. We can change this back once the bug is fixed.

Reply:
Oh and btw, I would have merged this PR, but I do not have the write access to do that.